src/main/clojure/clara/rules/compiler.clj

```diff
@@ -1242,19 +1246,191 @@
+                                                          (:env beta-node))
+                                     id
+                                     :join-filter-expr
+                                     {:compile-ctx {:condition condition
+                                                    :join-filter-expressions (:join-filter-expressions beta-node)
+                                                    :env (:env beta-node)
+                                                    :msg "compiling negation with join filter node"}})
+                        prev)
+            :test (handle-expr prev
+                               (compile-test (:constraints condition))
+                               id
+                               :test-expr
+                               {:compile-ctx {:condition condition
+                                              :env (:env beta-node)
+                                              :msg "compiling test node"}})
+            :accumulator (cond-> (handle-expr prev
+                                              (compile-accum (:accumulator beta-node) (:env beta-node))
+                                              id
+                                              :accum-expr
+                                              {:compile-ctx {:condition condition
+                                                             :accumulator (:accumulator beta-node)
+                                                             :env (:env beta-node)
+                                                             :msg "compiling accumulator"}})
+
+                                 (:join-filter-expressions beta-node)
+                                 (handle-expr (compile-join-filter (:join-filter-expressions beta-node)
+                                                                   (:join-filter-join-bindings beta-node)
+                                                                   (:new-bindings beta-node)
+                                                                   (:env beta-node))
+                                              id
+                                              :join-filter-expr
+                                              {:compile-ctx {:condition condition
+                                                             :join-filter-expressions (:join-filter-expressions beta-node)
+                                                             :env (:env beta-node)
+                                                             :msg "compiling accumulate with join filter node"}}))
+            :query prev)))
+      exprs
+      (:id-to-condition-node beta-graph))))
+
+(sc/defn compile-exprs :- schema/NodeFnLookup
+  "Takes a map in form produced by extract-exprs and evaluates the values(expressions) of the map in a batched manner.
+   This allows the eval calls to be more effecient, rather than evaluating each expression on its own."
```

Comment:
Maybe a line after this should say 
"See https://github.com/cerner/clara-rules/issues/381 for more details."